### ISSUE TITLE
Invalid ssl_certificate directive when --fullchain-path is missing

### ISSUE THREAD

**Issue by juniorz** [twstrike/letsencrypt#5]
Scenario:

``` gherkin
Given I already have a LE certificate
And I want to deploy it to a new nginx installation
When I run `letsencrypt install`
Then I should have all nginx files changed
And nginx should be accessible via HTTPS
```

This happens when I try to install a certificate previously obtained using `letsencrypt certonly`.

versions

``` console
root@dev:/home/letsencrypt/client# cat /etc/issue
Ubuntu 15.10 \n \l
root@dev:/home/letsencrypt/client# dpkg -s nginx | grep 'Version'
Version: 1.9.3-1ubuntu1
```

command

``` console
letsencrypt install --installer nginx \
  -d letsencrypt.you \
  --cert-path /etc/letsencrypt/live/letsencrypt.you/cert.pem \
  --key-path /etc/letsencrypt/live/letsencrypt.you/privkey.pem \
  --chain-path /etc/letsencrypt/live/letsencrypt.you/chain.pem \
  --text --debug -vvvvvvv
```

nginx config problem:

``` console
root@dev:~/letsencrypt-quickstart# nginx -t
nginx: [emerg] invalid number of arguments in "ssl_certificate" directive in /etc/nginx/nginx.conf:16
nginx: configuration file /etc/nginx/nginx.conf test failed
```

nginx config file diff (from default config files that ship with the package)

``` udiff
diff --git a/nginx.conf b/nginx.conf
index 01a4a21..86f34fc 100644
--- a/nginx.conf
+++ b/nginx.conf
@@ -6,9 +5,20 @@ events {
     worker_connections 768;
     # multi_accept on;
diff --git a/nginx.conf b/nginx.conf
index 01a4a21..86f34fc 100644
--- a/nginx.conf
+++ b/nginx.conf
@@ -6,9 +5,20 @@ events {
     worker_connections 768;
     # multi_accept on;
 }
-
 http {

+    server {
+        ssl_stapling_verify on;
+        ssl_stapling on;
+        ssl_trusted_certificate /etc/letsencrypt/live/letsencrypt.you/chain.pem;
+        include /etc/letsencrypt/options-ssl-nginx.conf;
+        ssl_certificate_key /etc/letsencrypt/live/letsencrypt.you/privkey.pem;
+        ssl_certificate;
+        error_log /var/lib/letsencrypt/error.log;
+        access_log /var/lib/letsencrypt/access.log;
+        listen 443 ssl;
+        server_name letsencrypt.you;
+    }
     ##
     # Basic Settings
     ##
@@ -29,10 +35,9 @@ http {
     ##
     # SSL Settings
     ##
-
-       ssl_protocols TLSv1 TLSv1.1 TLSv1.2; # Dropping SSLv3, ref: POODLE
+    ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
+    # Dropping SSLv3, ref: POODLE
     ssl_prefer_server_ciphers on;
-
     ##
     # Logging Settings
     ##
```

**Comment by juniorz** [comment]
Seems to be pretty similar to #3 but now the culprit is a missing `--fullchain-path` config. The command should be:

```
letsencrypt install --installer nginx \
  -d letsencrypt.you \
  --cert-path /etc/letsencrypt/live/letsencrypt.you/cert.pem \
  --key-path /etc/letsencrypt/live/letsencrypt.you/privkey.pem \
  --chain-path /etc/letsencrypt/live/letsencrypt.you/chain.pem \
  --fullchain-path /etc/letsencrypt/live/letsencrypt.you/fullchain.pem \
  --text --debug -vvvvvvv
```

But again, neither the client nor the plugin tells me this config is required.

Note: my original command seems to be wrong. I think it should use `--key-path /etc/letsencrypt/live/letsencrypt.you/privkey.pem`.

**Comment by juniorz** [comment]
This issue makes me wonder why can't we use convention over configuration in this plugin.

Why would somebody have all those files not located in `/etc/letsencrypt/live/<domain>/`? And if this is the case, would not a symlink to the default location be enough?

Is there any reasonable scenario for having to configure all of this individually?

I suspect, for security reasons, someone would have to place those files in different filesystems and a symlink would not be possible. Does it make sense?

**Comment by juniorz** [comment]
For now, improving the UX by failing on a missing config options seems to be the easy fix.

**Comment by tcz001** [comment]
Actually --chain-path --fullchain-path and --cert-path should not happen at the same time

there can be cases `--chain-path --cert-path` or `--fullchain-path`

**Comment by juniorz** [comment]
Is there any reference for this?
Em 04/01/2016 2:11 PM, "Fab Torchz J" notifications@github.com escreveu:

**Comment by tcz001** [comment]
so if you compare `fullchain.pem` and `cert.pem >> chain.pem`, they should be same

**Comment by tcz001** [comment]
And nginx will check cert against a trusted CA chain (when enabling the ssl_stapling_verify policy) if it's specified as `ssl_trusted_certificate` in configuration file,  
BUT remember if browser/client don't trust this chain, it will fail in the other side, so usually `ssl_trusted_certificate` with `ssl_stapling_verify on;` are for a self signed CA infrastructure in a private network.
http://nginx.org/en/docs/http/ngx_http_ssl_module.html

**Comment by juniorz** [comment]
This comment mentions you need the fullchain in the `ssl_certificate` config.

https://github.com/letsencrypt/letsencrypt/issues/1736#issuecomment-167327236

**Comment by juniorz** [comment]
For reference, all those issues are related to the current behavior of the nginx plugin. What I'm doing is trying to follow the `--help` and fixing all the misinformation in the tool.

Long story short: the nginx plugin may require parameters that are not mandatory for the `install` command, but the plugin does not warn you if you don't provide these parameters. This is the scope of my changes at the moment: you should be able to figure your way out.

**Comment by juniorz** [comment]
Fixed by ffc45e7.
